src/utils.zig

```diff
@@ -4,13 +4,58 @@ const serialize = lib.serialize;
 const deserialize = lib.deserialize;
 const isFixedSizeObject = lib.isFixedSizeObject;
 const ArrayList = std.ArrayList;
+const Allocator = std.mem.Allocator;
+const sha256 = std.crypto.hash.sha2.Sha256;
+const hashes_of_zero = @import("./zeros.zig").hashes_of_zero;
+
+// SSZ specification constants
+const BYTES_PER_CHUNK = lib.BYTES_PER_CHUNK;
+const chunk = lib.chunk;
+const zero_chunk = lib.zero_chunk;
+
+/// Returns true if the type is a utils.List type
+pub fn isListType(comptime T: type) bool {
+    if (@typeInfo(T) != .@"struct") return false;
+
+    // Primary: check for explicit SSZ type marker
+    if (@hasDecl(T, "ssz_type_kind")) {
+        return T.ssz_type_kind == .list;
+    }
```

Comment:
This might have been lost from my previous review, but I don't think you need to define `ssz_type_kind`. You should just be able to perform a `comptime` check that the bitlist is of type `utils.BitList`.